core/src/main/scala/com/banno/cosmos4s/CosmosBulkClient.scala

```diff
@@ -67,35 +67,43 @@ object CosmosBulkClient {
     import scala.collection.JavaConverters._
 
     def insert(value: List[Json]): F[Unit] =
-      Sync[F].delay(
-        executor.importAll(
-          value.map(_.noSpaces).asJava,
-          false,
-          true,
-          maxConcurrencyPerPartitionRange)) >>= { r =>
-        if (r.getNumberOfDocumentsImported() == value.size)
-          ().pure[F]
-        else
-          CosmosBulkInsertFailure(r).raiseError
+      Sync[F]
+        .delay(executor
+          .importAll(value.map(_.noSpaces).asJava, false, true, maxConcurrencyPerPartitionRange))
+        .map(Option(_)) >>= {
```

Comment:
From what I can tell It doesn't return `null`, it throws a `NullPointerException` internally?

Reply:
I saw the exception you mentioned, however I think we have this one as well